src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -710,8 +710,15 @@ public override BoundNode VisitCatchBlock(BoundCatchBlock node)
             var locals = node.Locals;
 
             var exceptionFilterPrologueOpt = node.ExceptionFilterPrologueOpt;
-            Debug.Assert(exceptionFilterPrologueOpt is null); // it is introduced by this pass
             BoundSpillSequenceBuilder builder = null;
+            if (exceptionFilterPrologueOpt != null)
```

Comment:
Did this PR change at what point exception filters are added? I didn't understand that part.

Reply:
It did not; what changed is where `SpillSequenceSpiller` can get called. Previously, it could never get called in a scenario where the prologue was used. Now it can; async try/finally rewriting can use it. So `SpillSequenceSpiller` can no longer assume that it will be empty.